src/main/java/io/jsonwebtoken/JwtParser.java

```diff
@@ -27,6 +28,52 @@
     public static final char SEPARATOR_CHAR = '.';
 
     /**
+     * Sets the expected value for the iss registered claim. If the actual content of the JWT does not match
+     * (including if it's empty), then an exception will be thrown
+     *
+     * @param issuer the string representing the expected value for the iss (issuer) claim of the JWT
+     * @return the parser for method chaining.
+     */
+    JwtParser setIssuer(String issuer);
```

Comment:
I think we need a `expect(String claimName, Object value)` base method that can be used to assert any claim - not just the standard ones.

Then, we can have convenience methods for the standard ones that delegate to that method.

After thinking about it, I'd probably also call these methods `expectIssuer`, `expectAudience`... etc.  The reason is that you're not really setting properties of the parser itself (which is what most setter methods are for - configuring the invoked object), but telling the parser to expect certain values.  

Thoughts?